### PR TITLE
Add a support for ignoring the events from a namespace that matches regexp

### LINKED ISSUE #295: Ignore namespaces using regex
**Is your feature request related to a problem? Please describe.**

I want to ignore multiple namespaces using a wildcard like: `project-experimental-*`

**Describe the solution you'd like**

It would be awesome if we can specify labels or regex to ignore short-lived namespaces.

**Describe alternatives you've considered**

None.

**Additional context**

I try using `project-experimental`  hoping that `string.Contains`  will match but it didn't.

### PR DESCRIPTION
**Pull request by jkremser**
Fixes issue #295

##### ISSUE TYPE

 - Feature Pull Request

##### SUMMARY

This change allows to ignore whole sets of namespaces using the asterix expansion. It still does the exact match check for all the namespaces (separated by comma in config), but if it contains a `*`, it's actually replaced with `.*` and passed to `regexp.MatchString` to do the job.

PR also contains couple of very simple test cases.

deps:
`namespace_checker.go` now imports also `regexp`

Fixes #295

It's my very first golang code, so please let me know if it's not idiomatic go :cookie:

### CODE REVIEW

**Comment by PrasadG193** [comment]
Awesome! Thanks for the PR @jkremser :pray: 
We should add an e2e test for this. Will file a separate ticket for that

**Review by PrasadG193 — APPROVED** [review]

**Review by PrasadG193 — CHANGES_REQUESTED** [review]

File: pkg/filterengine/filters/namespace_checker.go [hunk 1]
```diff
@@ -75,10 +76,20 @@ func (f NamespaceChecker) Describe() string {
...
+			for _, ignoredNamespace := range resourceNamespaces.Ignore {
+				// exact match
+				if ignoredNamespace == eventNamespace {
+					return true
+				}
+
+				// regexp
+				if strings.Contains(ignoredNamespace, "*") {
+					ns := strings.Replace(ignoredNamespace, "*", ".*", -1)
+					matched, _ := regexp.MatchString(ns, eventNamespace)
```
[the remark this answers is not in the document]
Comment (jkremser) [reply pkg/filterengine/filters/namespace_checker.go]: done + there was also a logical error I've fixed. Returning the `matched` boolean was not correct behavior.

File: pkg/filterengine/filters/namespace_checker.go [hunk 2]
```diff
@@ -75,9 +76,20 @@ func (f NamespaceChecker) Describe() string {
...
+				// exact match
+				if ignoredNamespace == eventNamespace {
+					return true
+				}
+
+				// regexp
+				if strings.Contains(ignoredNamespace, "*") {
+					ns := strings.Replace(ignoredNamespace, "*", ".*", -1)
+					matched, err := regexp.MatchString(ns, eventNamespace)
+					if matched && err == nil {
```
Comment (PrasadG193) [inline pkg/filterengine/filters/namespace_checker.go:89]:
```suggestion
					if err == nil && matched {
```

**Review by PrasadG193 — APPROVED** [review]

**Comment by chetanpdeshmukh** [comment]
Hello @jkremser ,
Thanks again for the PR!

Thank you for contributing to BotKube. Could you please fill out [this form](https://docs.google.com/forms/d/e/1FAIpQLSdULT22oZlpHPMXKVEtV4fZ6R8vdGDC2iFvMhoG6IhZkK1SFA/viewform?usp=sf_link), so we can send the well deserved awesome swags :slightly_smiling_face:

Team BotKube

**Pull request merged**